sklearn/linear_model/least_angle.py

```diff
@@ -427,6 +421,12 @@ def _lars_path_solver(X, y, Xy=None, Gram=None, n_samples=None, max_iter=500,
     elif copy_Gram:
         Gram = Gram.copy()
 
+    if X is not None and copy_X:
+        # force copy. setting the array to be fortran-ordered
+        # speeds up the calculation of the (partial) Gram matrix
+        # and allows to easily swap columns
+        X = X.copy('F')
```

Comment:
Without this change `test_lasso_lars_vs_R_implementation` breaks, meaning that `lars_path` doesn't seem to be respecting the `copy_X=True`.

Reply:
This might be an issue about requiring F order in the CD code, not doing inplace operations.